website/templates/project/contributors.mako

```diff
@@ -8,245 +8,365 @@
+                        notVisible: false
+                    }
+                }
+            },
+            inputs: {
+                nameSearch: '.name-search'
+            }
+        }">
+        <table  id="manageContributorsTable"
+                class="table responsive-table responsive-table-xxs"
+                data-bind="template: {
+                    name: 'contribTable',
+                    afterRender: afterRender,
+                    options: {
+                        containment: '#manageContributors'
+                    },
+                    data: 'contrib'
+                    }">
+        </table>
+    </div>
+    <div id="noContributors" data-bind="visible: $root.empty" class="no-items text-danger m-b-md">
+        No contributors found
+    </div>
+    <span id="adminContributorsAnchor" class="project-page anchor"></span>
+    <div id="adminContributors" data-bind="if: adminContributors().length">
+        <h4>
+            Admins on Parent Projects
+            <i class="fa fa-question-circle admin-info"
+                  data-content="These users are not contributors on
+                  this component but can view and register it because they
+                    are administrators on a parent project."
+                  data-toggle="popover"
+                  data-title="Admins on Parent Projects"
+                  data-container="body"
+                  data-placement="right"
+                  data-html="true"
+            ></i>
+        </h4>
+        <table  id="adminContributorsTable"
+                class="table responsive-table responsive-table-xxs"
+                data-bind="template: {
+                    name: 'contribTable',
+                    afterRender: afterRender,
+                    options: {
+                        containment: '#manageContributors'
+                    },
+                    data: 'admin'
+                }">
+        </table>
+        <div id="noAdminContribs" data-bind="visible: $root.adminEmpty" class="text-danger no-items m-b-md">
+            No administrators from parent project found.
+        </div>
+    </div>
+        ${buttonGroup()}
+    </div>
 
     % if 'admin' in user['permissions']:
-        <h3>View-only Links
-            <a href="#addPrivateLink" data-toggle="modal" class="btn btn-success btn-sm" style="margin-left:20px;margin-top: -3px">
+        <h3 class="m-t-xl">View-only Links
+            <a href="#addPrivateLink" data-toggle="modal" class="btn btn-success btn-sm m-l-md">
               <i class="fa fa-plus"></i> Add
             </a>
         </h3>
         <p>Create a link to share this project so those who have the link can view&mdash;but not edit&mdash;the project.</p>
         <div class="scripted" id="linkScope">
-
-            <table id="privateLinkTable" class="table">
-
+            <table id="privateLinkTable" class="table responsive-table responsive-table-xs"
+                    data-bind="if: visible">
```

Comment:
It's slightly preferable to use `visible` here, so that the `table` doesn't actually have to be repeatedly removed from the DOM